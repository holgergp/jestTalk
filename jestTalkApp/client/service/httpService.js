const getPosts = () => ({ posts: ['complicated', 'JSON', 'Response'] });

export default getPosts;
