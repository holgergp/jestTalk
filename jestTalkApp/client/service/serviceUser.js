import {getPosts} from '../service/httpService';

export let doSomethingWithRemoteData = () => {
    const postData  = getPosts();
    return postData.posts.map(it =>it.toUpperCase());
};