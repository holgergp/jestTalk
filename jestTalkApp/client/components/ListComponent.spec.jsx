import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import List from './ListComponent.jsx';
import * as dependency from '../service/httpService';

describe('ListComponent', () => {
  it('renders using mock fn', () => {
    dependency.getPosts = jest.fn();
    const testArray = ['1', '2'];
    const renderedList = renderer.create(<List values={testArray} />).toJSON();

    expect(renderedList).toMatchSnapshot();
    expect(dependency.getPosts).toHaveBeenCalled();
  });

  
  it('shallow renders', () => {
    const testArray = ['1', '2'];
    const renderer = new ShallowRenderer();
    const rendered = renderer.render(<List values={testArray} />);
    const shallowRendered = renderer.getRenderOutput();
    expect(shallowRendered).toMatchSnapshot();
  });
});
