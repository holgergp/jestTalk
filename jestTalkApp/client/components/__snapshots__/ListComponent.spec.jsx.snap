// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListComponent shallow renders 1`] = `
<ul>
  <li>
    <TextComponent
      value="1"
    />
  </li>
  <li>
    <TextComponent
      value="2"
    />
  </li>
</ul>
`;
